lib/AST/Decl.cpp

```diff
@@ -3734,11 +3734,9 @@ static Type computeNominalType(NominalTypeDecl *decl, DeclTypeKind kind) {
       // before the decl has been validated, so we rely on
       // the generic parameter list directly instead of looking
       // at the signature.
-      SmallVector<Type, 4> args;
-      for (auto param : decl->getGenericParams()->getParams())
-        args.push_back(param->getDeclaredInterfaceType());
-
-      return BoundGenericType::get(decl, ParentTy, args);
+      const auto sig = decl->getGenericSignature();
```

Comment:
This means calling getDeclaredInterfaceType() on a type will now compute it's generic signature. This may or may not be a problem. In any case, the above comment is out of date (it talks about "validation" which is no longer a thing).